jobqueue/src/main/java/com/path/android/jobqueue/JobQueue.java

```diff
@@ -95,6 +95,16 @@
             Collection<Long> excludeIds, String... tags);
 
     /**
+     * Returns all jobs not excluded by the conditions passed.
+     *
+     * @param excludeCancelled If true, cancelled jobs will not be returned. A job may be in cancelled
+     *                        state if a cancel request has arrived but it has not been removed from
+     *                        queue yet (e.g. still running).
+     * @param excludeIds Ids of jobs to ignore in the result
+     */
+    Set<JobHolder> findAllJobs(boolean excludeCancelled, Collection<Long> excludeIds);
```

Comment:
This is the API i'm planning to change.
I have not started yet but I'm thinking about changing these methods to receive a `Constraint` POJO. We may be able to merge some as well. Have not worked on it yet but thats next since now I have all tests passing :)